### DISCUSSION TITLE
Cross cutting dykes

### DISCUSSION THREAD

**Discussion by JOHANBLIGN** [gempy-project/gempy#584] [Q&A] [answered]
I am having trouble creating a simple model of a dyke/dike cross cutting a series of sedimentary layers. Tried adding the dyke to a different series with its own orientation, but the ordering is not right.  I can't get the non-conformal relationship on both sides of the dike. Haven't tried it yet, but I could potentially define to different series on both sides of the dike, but in scenarios with more complex structural geology with multiple intrusions, this will not be a good solution. Any suggestions will be appreciated! Sure I am missing something simple

**Comment by flohorovicic** [comment, marked as answer]
The easiest option is to generate several models - and then to combine the final results.

Here more details:
1) You could create a model without the dike, first
2) Then a model which _only_ contains the dike.
3) Then you can combine the final, generated models.

For the last point, the best procedure depends on the type of result you need. If you are interested in a voxel-model, then this step is quite easy: take the full model (1) and simply replace the cells which contain the dike from model (2).

We have been discussing a more general approach for a while - implementing a more hierarchical structure. For now, I would suggest looking at the workaround.

**Comment by Japhiolite** [comment]
Here's an example how this can be done to get a fault-zone:
https://github.com/Japhiolite/gempy-tests/blob/master/notebooks/Test-model_dyke_3D.ipynb

**Comment by Leguark** [comment]
Just adding that actual dyke support - with an easy to use interface - is something on the roadmap and hopefully will come online during 2021
